django/contrib/staticfiles/storage.py

```diff
@@ -84,12 +84,12 @@ def file_hash(self, name, content=None):
             md5.update(chunk)
         return md5.hexdigest()[:12]
 
-    def hashed_name(self, name, content=None):
+    def hashed_name(self, name, content=None, accept_fragments=False):
         parsed_name = urlsplit(unquote(name))
         clean_name = parsed_name.path.strip()
         opened = False
         if content is None:
-            if not self.exists(clean_name):
+            if not (accept_fragments or self.exists(clean_name)):
```

Comment:
1. If you mean that it generates file names without hashes *when using the 'partial' mode*, that's true. (Having found the file, it will generate hashed names normally.) But it should be up to the developer to understand this tradeoff. I might want to do it with a few files, but I may still chose to use chache busting for the majority of the static files. Also note that ``HashedFileMixin`` still makes the files accessable through the original, non-hashed name.

2. Exactly that's why the accept_fragments parameter is there and that's why it defaults to False. The collectstatic management command will call this method without specifying accept_fragments, so it will fail on missing files. (I don't know what some *strawman* developers do, nor do I deeply care about it ;) )